realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -89,7 +89,7 @@ protected BaseRealm(RealmConfiguration configuration, boolean autoRefresh) {
      * thread.
      *
      * @param autoRefresh {@code true} will turn auto-refresh on, {@code false} will turn it off.
-     * @throws IllegalStateException if called from a non-Looper thread.
+     * @throws IllegalStateException if called from a non-Looper thread with {@code autoRefresh} {@code true}.
```

Comment:
Wouldn't it make more sense to just throw an IllegalStateException always? This behavior seems a bit weird?